### ISSUE TITLE
OSX: KeyCtrl doesn't release when released in another app (e.g. when using Cmd+Tab)

### ISSUE THREAD

**Issue by jparismorgan** [ocornut/imgui#3832]
**Full info**
```
Dear ImGui 1.80 WIP (17911)
--------------------------------
sizeof(size_t): 8, sizeof(ImDrawIdx): 2, sizeof(ImDrawVert): 20
define: __cplusplus=201703
define: __APPLE__
define: __GNUC__=4
define: __clang_version__=12.0.0 (clang-1200.0.32.28)
define: IMGUI_HAS_VIEWPORT
define: IMGUI_HAS_DOCK
--------------------------------
io.BackendPlatformName: imgui_impl_osx
io.BackendRendererName: imgui_impl_opengl3
io.ConfigFlags: 0x00000040
 DockingEnable
io.ConfigViewportsNoDecoration
io.ConfigMacOSXBehaviors
io.ConfigInputTextCursorBlink
io.ConfigWindowsResizeFromEdges
io.ConfigWindowsMoveFromTitleBarOnly
io.ConfigMemoryCompactTimer = 60.0
io.BackendFlags: 0x0000100A
 HasMouseCursors
 RendererHasVtxOffset
 RendererHasViewports
--------------------------------
```

**Version/Branch of Dear ImGui:**

Version: 1.80 WIP - commit `388ca563dbb0fa683d2a8fbf1daaf2ab77f98a21`
Branch: docking

**Back-end/Renderer/Compiler/OS**

Back-ends: `imgui_impl_opengl3.h` + `imgui_impl_osx.h`
Compiler: `clang-1200.0.32.28`
Operating System: `OS X`

**My Issue/Question:**

When on a mac, you can use "Cmd + Tab" to quickly jump to another window. When you do that and then come back, `ImGui::GetIO().KeyCtrl` will be true even if you're not pressing down the `Cmd` key.

**Standalone, minimal, complete and verifiable example:** _(see https://github.com/ocornut/imgui/issues/2261)_
Run the following code. Then use "Cmd + Tab" to go to another window, then come back. You'll see that `KeyShift: true`.

```
void cmdRepro() {
    ImGuiIO &io = ImGui::GetIO();

    bool IsWindowAppearing = ImGui::IsWindowAppearing();
    bool IsWindowHovered = ImGui::IsWindowHovered();
    bool IsItemHovered = ImGui::IsItemHovered();
    bool RightDown = ImGui::IsMouseDown(ImGuiMouseButton_Right);
    bool KeySuper = io.KeySuper;
    bool KeyCtrl = io.KeyCtrl;
    bool KeyAlt = io.KeyAlt;
    bool KeyShift = io.KeyShift;

    ImGui::Text("IsWindowAppearing: %d IsWindowHovered: %d IsItemHovered: %d RightDown: %d KeySuper: %d KeyCtrl: %d KeyAlt: %d KeyShift: %d",
      IsWindowAppearing,
      IsWindowHovered, IsItemHovered, RightDown, KeySuper,
      KeyCtrl,
      KeyAlt,
      KeyShift
    );
}
```

**Workaround:**
You can manually reset it with:
```
    // If you cmd + tab to another window then come back io.KeySuper will be true. Reset it.
    if (ImGui::IsWindowAppearing()) {
      ImGui::GetIO().KeySuper = false;
    }
```

**Comment by ocornut** [comment]
> you can use "Cmd + Tab" to quickly jump to another window. 

I don't know what Cmd+Tab is mapped to. It is switching to other OS window (OSX ui) or other Dear ImGui window? (dear imgui "ctrl+tab" window). Given your ConfigFlags I am assuming keyboard nav is disabled so it is the earlier, but it is bizarre you suggest using `IsWindowAppearing()` which is testing for Dear ImGui windows appearing whereas the suggestion relate to OS window gaining/losing focus.

I believe this is the same as #3532 but specifically for the OSX backend.

**Comment by jparismorgan** [comment]
> > you can use "Cmd + Tab" to quickly jump to another window.
> 
> I don't know what Cmd+Tab is mapped to. It is switching to other OS window (OSX ui) or other Dear ImGui window? (dear imgui "ctrl+tab" window). Given your ConfigFlags I am assuming keyboard nav is disabled so it is the earlier, but it is bizarre you suggest using `IsWindowAppearing()` which is testing for Dear ImGui windows appearing whereas the suggestion relate to OS window gaining/losing focus.
> 
> I believe this is the same as #3532 but specifically for the OSX backend.

Good question. It's switching to other OS window (OSX ui). I also tested more and you're correct, using `IsWindowAppearing()` is not a proper fix for this, it still happens.

**Comment by ocornut** [comment]
I was referring to _back-end_ clearing all Keys[] and key modifiers when all dear imgui windows are unfocused.
This is probably the better default solution to implement across all back-ends, which maybe an option to disable it.

**Comment by rokups** [comment]
What if you alt-tab to another window, keep holding alt and then alt-tab back? No new "key-down" event would fire, alt would remain held down but down state would be cleared as window regained focus.

**Comment by ocornut** [comment]
> What if you alt-tab to another window, keep holding alt and then alt-tab back? No new "key-down" event would fire, alt would remain held down but down state would be cleared as window regained focus.

That's technically a bug but pragmatically this is less problematic and a MUCH better default than currently and many apps behave this way. I'll now be applying this clearing in the Win32 backend and would appreciate the equivalent PR for native OSX backend.

**Comment by ocornut** [comment]
Merged fix by @rokups: 6d5388448794f76cc92576a74ad3cb17da2fca84
